src/lib.rs

```diff
@@ -65,124 +67,147 @@ pub fn program_from_file(bin_path: &str) -> Vec<U256> {
-                    LogOpcode::TransientStorageRead => _transient_storage_read(&mut vm, &opcode),
-                    LogOpcode::TransientStorageWrite => _transient_storage_write(&mut vm, &opcode),
+                    LogOpcode::TransientStorageRead => {
+                        _transient_storage_read(&mut vm, &opcode)?;
+                    }
+                    LogOpcode::TransientStorageWrite => {
+                        _transient_storage_write(&mut vm, &opcode)?;
+                    }
                 },
-                Variant::FarCall(far_call_variant) => far_call(&mut vm, &far_call_variant),
+                Variant::FarCall(far_call_variant) => far_call(&mut vm, &far_call_variant)?,
                 // TODO: This is not how return works. Fix when we have calls between contracts
                 // hooked up.
                 // This is only to keep the context for tests
                 Variant::Ret(ret_variant) => match ret_variant {
                     RetOpcode::Ok => {
-                        let should_break = _ok(&mut vm, &opcode);
+                        let should_break = _ok(&mut vm, &opcode)?;
                         if should_break {
                             break;
                         }
                     }
                     RetOpcode::Revert => {
-                        let should_break = _revert(&mut vm, &opcode);
+                        let should_break = _revert(&mut vm, &opcode)?;
                         if should_break {
                             panic!("Contract Reverted");
                         }
                     }
                     RetOpcode::Panic => {
-                        let should_break = _panic(&mut vm, &opcode);
+                        let should_break = _panic(&mut vm, &opcode)?;
                         if should_break {
                             panic!("Contract Panicked");
                         }
                     }
                 },
 
                 Variant::UMA(uma_variant) => match uma_variant {
-                    UMAOpcode::HeapRead => _heap_read(&mut vm, &opcode),
-                    UMAOpcode::HeapWrite => _heap_write(&mut vm, &opcode),
-                    UMAOpcode::AuxHeapRead => _aux_heap_read(&mut vm, &opcode),
-                    UMAOpcode::AuxHeapWrite => _aux_heap_write(&mut vm, &opcode),
-                    UMAOpcode::FatPointerRead => _fat_pointer_read(&mut vm, &opcode),
+                    UMAOpcode::HeapRead => _heap_read(&mut vm, &opcode)?,
+                    UMAOpcode::HeapWrite => _heap_write(&mut vm, &opcode)?,
+                    UMAOpcode::AuxHeapRead => _aux_heap_read(&mut vm, &opcode)?,
+                    UMAOpcode::AuxHeapWrite => _aux_heap_write(&mut vm, &opcode)?,
+                    UMAOpcode::FatPointerRead => _fat_pointer_read(&mut vm, &opcode)?,
                     UMAOpcode::StaticMemoryRead => todo!(),
                     UMAOpcode::StaticMemoryWrite => todo!(),
```

Comment:
If we want to avoid crashes we may want to add an `_unimplemented()` helper that matches the signature of other opcodes and reverts and does proper error reporting.

Reply:
I'm not sure if it is a needed change, as we are running the vm only with test which only use instructions already implemented. 

It may be useful tho for when we get further in development of the virtual machine, and run programs that are not from our tests.